### PR TITLE
[CSV] Progress Bar for compressed files.

### PR DESCRIPTION
**Pull request by pdet**
When scanning compressed files, we use a different strategy to check for progress.

Instead of counting the bytes our CSV scanners have scanned from the file and comparing that to the total file size, we use the file handle to tell us how far we are on that file.

### CODE REVIEW

**Comment by rustyconover** [comment]
How will this affect the use of reading input from pipes?  With pipes there isn't a concept of length.

File: src/execution/operator/csv_scanner/buffer_manager/csv_file_handle.cpp [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
 CSVFileHandle::CSVFileHandle(FileSystem &fs, Allocator &allocator, unique_ptr<FileHandle> file_handle_p,
                              const string &path_p, FileCompressionType compression)
-    : file_handle(std::move(file_handle_p)), path(path_p) {
+    : compression_type(compression), file_handle(std::move(file_handle_p)), path(path_p) {
 	can_seek = file_handle->CanSeek();
 	on_disk_file = file_handle->OnDiskFile();
 	file_size = file_handle->GetFileSize();
 	is_pipe = file_handle->IsPipe();
-	uncompressed = compression == FileCompressionType::UNCOMPRESSED;
+	if (compression_type == FileCompressionType::AUTO_DETECT) {
```
Comment (Mytherin) [inline src/execution/operator/csv_scanner/buffer_manager/csv_file_handle.cpp:16]: This seems to duplicate logic that is in `VirtualFileSystem::OpenFile` - could we perhaps make `GetCompression()` a virtual method on the `FIleHandle` that defaults to `FileCompressionType::UNCOMPRESSED`?

File: extension/parquet/zstd_file_system.cpp [hunk 2]
```diff
@@ -160,6 +160,10 @@ class ZStdFile : public CompressedFile {
 		Initialize(write);
 	}
 
+	DUCKDB_API FileCompressionType GetFileCompressionType() override {
```
Comment (Mytherin) [inline extension/parquet/zstd_file_system.cpp:163]: `DUCKDB_API` needs to be removed here

File: src/common/gzip_file_system.cpp [hunk 3]
```diff
@@ -300,7 +300,9 @@ class GZipFile : public CompressedFile {
 	    : CompressedFile(gzip_fs, std::move(child_handle_p), path) {
 		Initialize(write);
 	}
-
+	DUCKDB_API FileCompressionType GetFileCompressionType() override {
```
Comment (Mytherin) [inline src/common/gzip_file_system.cpp:303]: Same here

File: src/common/file_system.cpp [hunk 4]
```diff
@@ -623,6 +627,10 @@ FileType FileHandle::GetType() {
 	return file_system.GetFileType(*this);
 }
 
+DUCKDB_API idx_t FileHandle::GetProgress() {
```
Comment (Mytherin) [inline src/common/file_system.cpp:630]: Same here

**Review by Mytherin — COMMENTED** [review]
Looks like this breaks the Windows build, most likely because of the extra `DUCKDB_API` labels. `DUCKDB_API` should only be used in headers and only for non-inlined functions.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**